src/voting-strategies/OtterspaceBadgesVotingStrategy.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.18;
+
+// import { Badges } from "@otterspace/contracts/Badges.sol";
+import { IVotingStrategy } from "../interfaces/IVotingStrategy.sol";
+
+interface IBadges {
+    function ownerOf(uint256 tokenId) external view returns (address);
+}
+
+/// @title Otterspace Badge Voting Strategy
+/// @notice Allows Otterspace Badges to be used for voting power.
+contract OtterspaceBadgesVotingStrategy is IVotingStrategy {
+    error InvalidBadge();
+
+    /// @notice The address of the Otterspace Badges Contract.
+    address public badgesRegistry;
+
+    /// @dev Data stored as parameters for each Badge.
+    struct Badge {
+        // spec of the badge.
+        string specUri;
+        // The voting power granted to badges of this spec.
+        uint96 vp;
+    }
+
+    constructor(address BadgesRegistry) {
+        badgesRegistry = BadgesRegistry;
+    }
+
+    /// @notice Returns the voting power of an address.
+    /// @param voter The address to get the voting power of.
+    /// @param params Parameter array containing the encoded whitelist of addresses and their voting power.
+    ///               The array should be an ABI encoded array of Member structs.
+    /// @param userParams Parameter array containing an array of indices of
```

Comment:
Is this required? Can't we just query with balanceOf or something?

Reply:
i thought we wanted different voting power for different badges? that would just give a single balance of all badges the address owns.